Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp

```diff
@@ -216,54 +216,56 @@ void SVGFEConvolveMatrixElement::svgAttributeChanged(const QualifiedName& attrNa
 
 RefPtr<FilterEffect> SVGFEConvolveMatrixElement::createFilterEffect(const FilterEffectVector&, const GraphicsContext&) const
 {
-    int orderXValue = orderX();
-    int orderYValue = orderY();
-    if (!hasAttribute(SVGNames::orderAttr)) {
-        orderXValue = 3;
-        orderYValue = 3;
+    int orderXValue = 3;
+    int orderYValue = 3;
+    if (hasAttribute(SVGNames::orderAttr)) {
+        orderXValue = orderX();
+        orderYValue = orderY();
+        // Spec says order must be > 0. Bail if it is not.
+        if (orderXValue <= 0 || orderYValue <= 0)
+            return nullptr;
     }
-    // Spec says order must be > 0. Bail if it is not.
-    if (orderXValue < 1 || orderYValue < 1)
-        return nullptr;
 
     auto& kernelMatrix = this->kernelMatrix();
-    int kernelMatrixSize = kernelMatrix.items().size();
+    size_t kernelMatrixSize = kernelMatrix.items().size();
     // The spec says this is a requirement, and should bail out if fails
-    if (orderXValue * orderYValue != kernelMatrixSize)
-        return nullptr;
-
-    if (!isValidTargetXOffset())
+    if ((size_t)(orderXValue * orderYValue) != kernelMatrixSize)
         return nullptr;
 
-    if (!isValidTargetYOffset())
+    if (!isValidTargetXOffset() || !isValidTargetYOffset())
         return nullptr;
 
-    int targetXValue = targetX();
+    int targetXValue;
     // The spec says the default value is: targetX = floor ( orderX / 2 ))
-    if (!hasAttribute(SVGNames::targetXAttr))
+    if (hasAttribute(SVGNames::targetXAttr))
         targetXValue = static_cast<int>(floorf(orderXValue / 2));
+    else
+        targetXValue = targetX();
 
-    int targetYValue = targetY();
+    int targetYValue;
     // The spec says the default value is: targetY = floor ( orderY / 2 ))
-    if (!hasAttribute(SVGNames::targetYAttr))
+    if (hasAttribute(SVGNames::targetYAttr))
         targetYValue = static_cast<int>(floorf(orderYValue / 2));
+    else
+        targetYValue = targetY();
 
```

Comment:
The logic here isn't equivalent, you inverted the if / else here.

Side note: I think this would look better written as a ternary:

```
// The spec says the default value is: targetX = floor ( orderX / 2 ))
int targetXValue = hasAttribute(SVGNames::targetXAttr) ? targetX() : static_cast<int>(floorf(orderXValue / 2));

// The spec says the default value is: targetY = floor ( orderY / 2 ))
int targetYValue = hasAttribute(SVGNames::targetYAttr) ? targetY() : static_cast<int>(floorf(orderYValue / 2));

Reply:
Fixed!